libtelemetry/src/main/java/com/mapbox/services/android/telemetry/PermissionCheckRunnable.java

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.services.android.telemetry;
+
+import android.content.Context;
+import android.os.Handler;
+
+import com.mapbox.services.android.core.permissions.PermissionsManager;
+
+class PermissionCheckRunnable implements Runnable {
+  private final Context context;
+  private final Handler handler = new Handler();
+  private final ExponentialBackoff counter = new ExponentialBackoff();
+  private MapboxTelemetry mapboxTelemetry;
+
+  PermissionCheckRunnable(Context context, MapboxTelemetry mapboxTelemetry) {
+    this.context = context;
+    this.mapboxTelemetry = mapboxTelemetry;
+  }
+
+  @Override
+  public void run() {
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
+      mapboxTelemetry.locationAndReceiverSetup();
```

Comment:
If `isTelemetryEnabled` or `isOpted` change their value between intervals will cause problems.
No need to create `locationAndReceiverSetup` method because we could use the `public` method `optIn` which will check those variables too.